xline/src/storage/kvstore.rs

```diff
@@ -221,10 +246,35 @@ impl KvStoreInner {
         kvs
     }
 
+    /// Get `KeyValue` start from a revision and convert to `Event`
+    pub(crate) fn get_event_from_revision(&self, key_range: KeyRange, revision: i64) -> Vec<Event> {
+        let revisions = match key_range.end.as_slice() {
+            ONE_KEY => self.index.get_one_from_rev(&key_range.start, revision),
+            ALL_KEYS => self.index.get_all_from_rev(revision),
+            _ => self.index.get_range_from_rev(key_range, revision),
+        };
+        let values = self.db.get_values(&revisions);
+        values
+            .into_iter()
+            .map(|kv| {
+                // Delete
+                #[allow(clippy::as_conversions)] // This cast is always valid
+                let event_type = if kv.version == 0 && kv.create_revision == 0 {
+                    EventType::Delete as i32
+                } else {
+                    EventType::Put as i32
+                };
+                Event {
+                    r#type: event_type,
```

Comment:
Why can't we use type `EventType` for the field `r#type`?

Reply:
In `Event`, `r#type` filed is i32, we need to cast.

Reply:
Fix this cast issue.